src/core/comm/coll.cc

```diff
@@ -291,7 +294,10 @@ int collFinalize(void)
   mpi_comms.clear();
   int fina_flag = 0;
   CHECK_MPI(MPI_Finalized(&fina_flag));
-  if (fina_flag == 0) { CHECK_MPI(MPI_Finalize()); }
+  if (fina_flag == 1) {
+    log_coll.fatal("MPI should not been finalized");
```

Comment:
Typo: `"MPI should not have been finalized"`

Reply:
Thanks, it should have been fixed now.